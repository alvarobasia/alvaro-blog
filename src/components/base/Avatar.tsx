import Image from 'next/image';
import {useState} from "react"

interface AvatarProps {
    src: string;
}

const Avatar: React.FC<AvatarProps> = (props:AvatarProps) => {

  return <div className='overflow-hidden w-[100px] h-[100px] rounded-full bg-slate-700 border-2 border-white'>
    <Image src={props.src} width={100} height={100} alt="Imagem do desenvolvedor do blog"/>
  </div>;
};

export default Avatar;
