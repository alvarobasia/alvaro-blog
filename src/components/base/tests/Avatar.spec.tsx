import React from 'react';

import {render, screen} from '@testing-library/react';

import Avatar from '../Avatar';

describe('Test Avatar component', () => {
  it('should render Avatar component', () => {
    const {baseElement} = render(
      <Avatar src="https://github.com/alvarobasia.png" />
    );
    const avatar = screen.getByRole('img');
    expect(avatar).toHaveAttribute(
      'src',
      '/_next/image?url=https%3A%2F%2Fgithub.com%2Falvarobasia.png&w=256&q=75'
    );
    expect(baseElement).toMatchSnapshot();
  });

  it('should show correct image in Avatar', () => {
    render(<Avatar src="https://github.com/alvarobasia.png" />);
    const avatar = screen.getByRole('img');
    expect(avatar).toHaveAttribute(
      'src',
      '/_next/image?url=https%3A%2F%2Fgithub.com%2Falvarobasia.png&w=256&q=75'
    );
  });
});
