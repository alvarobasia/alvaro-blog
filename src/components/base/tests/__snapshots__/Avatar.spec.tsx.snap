// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test Avatar component > should render Avatar component 1`] = `
<body>
  <div>
    <div
      class="overflow-hidden w-[100px] h-[100px] rounded-full bg-slate-700 border-2 border-white"
    >
      <img
        alt="Imagem do desenvolvedor do blog"
        data-nimg="1"
        decoding="async"
        height="100"
        loading="lazy"
        src="/_next/image?url=https%3A%2F%2Fgithub.com%2Falvarobasia.png&w=256&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fgithub.com%2Falvarobasia.png&w=128&q=75 1x, /_next/image?url=https%3A%2F%2Fgithub.com%2Falvarobasia.png&w=256&q=75 2x"
        style="color: transparent;"
        width="100"
      />
    </div>
  </div>
</body>
`;

exports[`Test Avatar component > should render Avatar component 2`] = `
<body>
  <div>
    <div
      class="overflow-hidden w-[100px] h-[100px] rounded-full bg-slate-700 border-2 border-white"
    >
      <img
        alt="Imagem do desenvolvedor do blog"
        data-nimg="1"
        decoding="async"
        height="100"
        loading="lazy"
        src="/_next/image?url=https%3A%2F%2Fgithub.com%2Falvarobasia.png&w=256&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fgithub.com%2Falvarobasia.png&w=128&q=75 1x, /_next/image?url=https%3A%2F%2Fgithub.com%2Falvarobasia.png&w=256&q=75 2x"
        style="color: transparent;"
        width="100"
      />
    </div>
  </div>
</body>
`;
